model/src/pyrenew/convolve.py

```diff
@@ -17,62 +17,110 @@
+        X(t - n + 1), ... X(t - 1)\right] \dot \vec{d} \right)
+
+    Where `math`:\vec{d} is a vector of length `math`:n,
+    `math`:m(t) is a scalar for each value of time `math`:t,
+    and `math`:f is a scalar-valued function.
+
+    Given `math`:\vec{d}, and optionally `math`:f,
+    this factory function returns a new function that
+    peforms one step of this process while scanning along
+    an array of  multipliers (i.e. an array
+    giving the values of `math`:m(t)) using :py:func:jax.lax.scan.
     """
+    if transform is None:
+        transform = IdentityTransform()
 
     def _new_scanner(
         history_subset: ArrayLike, multiplier: float
     ) -> tuple[ArrayLike, float]:  # numpydoc ignore=GL08
-        new_val = multiplier * jnp.dot(discrete_dist_flipped, history_subset)
+        new_val = transform(
+            multiplier * jnp.dot(discrete_dist_flipped, history_subset)
+        )
         latest = jnp.hstack([history_subset[1:], new_val])
         return latest, new_val
 
     return _new_scanner
 
 
-def new_double_scanner(
+def new_double_convolve_scanner(
     dists: tuple[ArrayLike, ArrayLike],
-    transforms: tuple[Callable, Callable],
+    transforms: tuple[Callable, Callable] = (None, None),
 ) -> Callable:
     """
-    Factory function to create a scanner function that applies two sequential transformations
-    and convolutions using two discrete distributions.
+    Factory function to create a scanner function
+    that iteratively constructs arrays by applying
+    the dot-product/multiply/transform operation
+    twice per history subset, with the first yielding
+    operation yielding an additional scalar multiplier
+    for the second.
 
     Parameters
     ----------
     dists : tuple[ArrayLike, ArrayLike]
-        A tuple of two 1D jax arrays, each representing a discrete distribution for the
+        A tuple of two 1D jax arrays, each representing a
+        discrete distribution for the
         two stages of convolution.
     transforms : tuple[Callable, Callable]
-        A tuple of two functions, each transforming the output of the dot product at each
-        convolution stage.
+        A tuple of two functions, each transforming the
+        output of the dot product at each
+        convolution stage. If either is None,
+        the identity transformation will be used
+        at that step. Default (None, None)
 
     Returns
     -------
     Callable
-        A scanner function that applies two sequential convolutions and transformations.
-        It takes a history subset and a tuple of multipliers, computes the transformations
-        and dot products, and returns the updated history subset and a tuple of new values.
+        A scanner function that applies two sets of
+        convolution, multiply, and transform operations
+        in sequence to construct a new array by scanning
+        along a pair of input arrays that are equal in
+        length to each other.
     """
```

Comment:
A little bit of detail via `Notes` could be nice giving a math description as you are doing in `new_convolve_scanner()`.

Reply:
Done in 8c4a098